x-pack/plugins/security_solution/server/lib/app_features/security_assistant_kibana_features.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/server';
+import type { AppFeaturesAssistantConfig, BaseKibanaFeatureConfig } from './types';
+import { APP_ID, ASSISTANT_FEATURE_ID } from '../../../common/constants';
+import { AppFeatureAssistantKey } from '../../../common/types/app_features';
+import { AssistantSubFeatureId } from './security_assistant_kibana_sub_features';
+
+export const getAssistantBaseKibanaFeature = (): BaseKibanaFeatureConfig => ({
+  id: ASSISTANT_FEATURE_ID,
```

Comment:
So I did some additional testing and have a question around how this should work with `minimumLicense: 'enterprise'` configured. Steps were:

1. Start es/kibana from `main`, verify assistant functionality 👍 
2. Downgrade license to `basic`, assistant shows 'license upgrade' flow 👍 
3. Kill kibana, switch to `assistant-rbac` branch, start kibana...
4. Noticed `securitySolutionAssistant.isAssistantEnabled` is returning `true` (matching the role's `Security:All` privilege), however the license is `basic`, and with `minimumLicense: 'enterprise'` configured I am expecting this privilege to be `false` since the license requirement isn't met.

Does `minimumLicense` only lock out the UI, or is it supposed to have an affect here as well?

Reply:
> @thomheymann, one question here for you. Do we need to add any additional attributes here to ensure that roles configured on previous versions would have the `all` privilege for this new assistant privilege so long as the main Kibana feature has `all`, or does that happen automatically? I recall hearing that `includeIn: 'all'` was necessary for defaulting new privileges, but can't find if that's just for sub-features, or the main privilege category as well?

Existing roles configured with global privileges (called "Privileges for all features" in the role management screen) will automatically get all/read to newly introduced feature privileges. 

Existing roles with custom privileges (called "Customise by feature" in the UI) will not automatically have access to newly introduced roles.

Reply:
Roger 👍

So is there a way to set up the new privilege as `all` if the user has `all` or `read` for the security app (so 'Customize by feature') but not `all` global privileges ('Privileges for all features')? Or more generically, is there a way to specify a function to compute the initial value based off other privileges when introducing a new one like this? 

This would be ideal as it allows folks who currently have access to maintain access on upgrade, but if not currently possible, no worries -- we can make a note in the documentation.